exchanges/binance/binance_wrapper.go

```diff
@@ -2072,57 +2077,120 @@ func (b *Binance) GetServerTime(ctx context.Context, ai asset.Item) (time.Time,
 	return time.Time{}, fmt.Errorf("%s %w", ai, asset.ErrNotSupported)
 }
 
-// GetLatestFundingRate returns the latest funding rate for a given asset and currency
-func (b *Binance) GetLatestFundingRate(ctx context.Context, r *fundingrate.LatestRateRequest) (*fundingrate.LatestRateResponse, error) {
+// GetLatestFundingRates returns the latest funding rates data
+func (b *Binance) GetLatestFundingRates(ctx context.Context, r *fundingrate.LatestRateRequest) ([]fundingrate.LatestRateResponse, error) {
 	if r == nil {
 		return nil, fmt.Errorf("%w LatestRateRequest", common.ErrNilPointer)
 	}
 	if r.IncludePredictedRate {
 		return nil, fmt.Errorf("%w IncludePredictedRate", common.ErrFunctionNotSupported)
 	}
-	format, err := b.GetPairFormat(r.Asset, true)
-	if err != nil {
-		return nil, err
-	}
-	fPair := r.Pair.Format(format)
-	pairRate := fundingrate.LatestRateResponse{
-		Exchange: b.Name,
-		Asset:    r.Asset,
-		Pair:     fPair,
+	fPair := r.Pair
+	var err error
+	if !fPair.IsEmpty() {
+		var format currency.PairFormat
+		format, err = b.GetPairFormat(r.Asset, true)
+		if err != nil {
+			return nil, err
+		}
+		fPair = r.Pair.Format(format)
 	}
+
 	switch r.Asset {
 	case asset.USDTMarginedFutures:
 		var mp []UMarkPrice
-		mp, err = b.UGetMarkPrice(ctx, r.Pair)
+		mp, err = b.UGetMarkPrice(ctx, fPair)
 		if err != nil {
 			return nil, err
 		}
-		pairRate.TimeOfNextRate = time.UnixMilli(mp[len(mp)-1].NextFundingTime)
-		pairRate.LatestRate = fundingrate.Rate{
-			Time: time.UnixMilli(mp[len(mp)-1].Time).Truncate(b.Features.Supports.FuturesCapabilities.FundingRateFrequency),
-			Rate: decimal.NewFromFloat(mp[len(mp)-1].LastFundingRate),
+		resp := make([]fundingrate.LatestRateResponse, 0, len(mp))
+		for i := range mp {
+			var cp currency.Pair
+			var isEnabled bool
+			cp, isEnabled, err = b.MatchSymbolCheckEnabled(mp[i].Symbol, r.Asset, true)
+			if err != nil && !errors.Is(err, currency.ErrPairNotFound) {
+				return nil, err
+			}
+			if !isEnabled {
+				continue
+			}
+			var isPerp bool
+			isPerp, err = b.IsPerpetualFutureCurrency(r.Asset, cp)
+			if err != nil {
+				return nil, err
+			}
+			if !isPerp {
+				continue
+			}
+			nft := time.UnixMilli(mp[i].NextFundingTime)
+			rate := fundingrate.LatestRateResponse{
+				TimeChecked: time.Now(),
+				Exchange:    b.Name,
+				Asset:       r.Asset,
+				Pair:        cp,
+				LatestRate: fundingrate.Rate{
+					Time: time.UnixMilli(mp[i].Time).Truncate(b.Features.Supports.FuturesCapabilities.FundingRateFrequency),
```

Comment:
Binance has pairs which differ from the default 8 hour funding rate frequency which can be observed at this endpoint: https://binance-docs.github.io/apidocs/futures/en/#get-funding-rate-info - the default indeed is 8 hours however it can be overrided to 4 hours. GateIO does this also

Reply:
darn it!

Reply:
Just a note for this, the endpoint works for USDTMargined futures, but only returns a [] for CoinMargined, even though its documented as an endpoint.
The code will now parse using that endpoints data, but if its missing, I cannot rely on my own local calculations with the funding rate payment variance and it will just render whatever `mp[i].Time` is